src/connectService.ts

```diff
@@ -249,25 +266,97 @@ class ConnectService extends IframeService {
     return fromHex(seed.slice(2))
   }
 
-  async migrate3BoxProfile(did: string): Promise<void> {
+  // Returns 3box original profile
+  async migrate3BoxProfile(did: string): Promise<any> {
     assert.isDefined(this.idx, 'IDX instance must be defined')
-    assert.isDefined(this.ceramic, 'Ceramic instance must be defined')
-    assert.isDefined(this.threeId, 'ThreeIdProvider instance must be defined')
-
     const profile = await get3BoxProfile(did)
     const transform = transformProfile(profile)
     const existing = (await this.idx.get('basicProfile')) || {}
-    console.log(Object.assign(existing, transform))
     await this.idx.set('basicProfile', Object.assign(existing, transform))
```

Comment:
`await idx.merge('basicProfile', transform)` could be used instead here, as it implements this logic of load and shallow merge

Reply:
oh nice, yeah will start using that instead